--- lex.c
+++ lex.c
@@ -1084,7 +1084,7 @@
 
 	if (have_tty && ksh_tmout) {
 		ksh_tmout_state = TMOUT_READING;
-		alarm(ksh_tmout);
+		/* myos: no alarm(2) */
 	}
 	if (have_tty && (0
 #ifdef VI
@@ -1139,7 +1139,7 @@
 	source = s;
 	if (have_tty && ksh_tmout) {
 		ksh_tmout_state = TMOUT_EXECUTING;
-		alarm(0);
+		/* myos: no alarm(2) */
 	}
 	s->start = s->str = Xstring(s.xs, xp);
 	strip_nuls(Xstring(s.xs, xp), Xlength(s.xs, xp));
@@ -1282,24 +1282,17 @@
 				strbuf[1] = '\0';
 				break;
 			case 'h':	/* '\' 'h' shortened hostname */
-				gethostname(strbuf, sizeof strbuf);
-				p = strchr(strbuf, '.');
-				if (p)
-					*p = '\0';
+				strlcpy(strbuf, "myos", sizeof strbuf);
 				break;
 			case 'H':	/* '\' 'H' full hostname */
-				gethostname(strbuf, sizeof strbuf);
+				strlcpy(strbuf, "myos", sizeof strbuf);
 				break;
 			case 'j':	/* '\' 'j' number of jobs */
 				snprintf(strbuf, sizeof strbuf, "%d",
 				    j_njobs());
 				break;
 			case 'l':	/* '\' 'l' basename of tty */
-				p = ttyname(0);
-				if (p)
-					p = basename(p);
-				if (p)
-					strlcpy(strbuf, p, sizeof strbuf);
+				strlcpy(strbuf, "console", sizeof strbuf);
 				break;
 			case 'n':	/* '\' 'n' newline */
 				strbuf[0] = '\n';
